test/integration/toolchains/gnu/test_basic_layout.py

```diff
@@ -1,13 +1,16 @@
 import os
 import platform
 import textwrap
+import pytest
 
 from conan.test.utils.tools import TestClient
 
-
-def test_basic_layout_subproject():
+@pytest.mark.parametrize("basic_layout, expected_path", [
+    ('basic_layout(self)', 'build-release'),
+    ('basic_layout(self, build_folder="custom_build_folder")', 'custom_build_folder')])
+def test_basic_layout_subproject(basic_layout, expected_path):
     c = TestClient()
-    conanfile = textwrap.dedent("""
+    conanfile = textwrap.dedent(f"""
         from conan import ConanFile
         from conan.tools.layout import basic_layout
         class Pkg(ConanFile):
```

Comment:
Could we add the `settings = "build_type"` to check that the `custom_build_folder` is not adding that suffix?